pkg/fake/ec2api.go

```diff
@@ -316,10 +317,29 @@ func (e *EC2API) DescribeImagesWithContext(_ context.Context, input *ec2.Describ
 	if !e.DescribeImagesOutput.IsNil() {
 		return e.DescribeImagesOutput.Clone(), nil
 	}
+	if aws.StringValue(input.Filters[0].Name) == "image-id" {
```

Comment:
Have we thought about doing a similar filtering mechanisms to [how we do filtering with describeInstances](https://github.com/aws/karpenter/blob/main/pkg/fake/ec2api.go#L265). This way it more closely aligns with how the upstream API performs